src/Command/AnalyseApplication.php

```diff
@@ -29,24 +30,30 @@ class AnalyseApplication
 	/** @var \PHPStan\File\FileHelper */
 	private $fileHelper;
 
+	/** @var \PHPStan\Command\ErrorFormatter\ErrorFormatter */
+	private $errorFormatter;
+
 	public function __construct(
 		Analyser $analyser,
 		string $memoryLimitFile,
 		FileHelper $fileHelper,
-		array $fileExtensions
+		array $fileExtensions,
+		ErrorFormatter $errorFormatter
 	)
 	{
 		$this->analyser = $analyser;
 		$this->memoryLimitFile = $memoryLimitFile;
 		$this->fileExtensions = $fileExtensions;
 		$this->fileHelper = $fileHelper;
+		$this->errorFormatter = $errorFormatter;
 	}
 
 	/**
 	 * @param string[] $paths
 	 * @param \Symfony\Component\Console\Style\StyleInterface $style
 	 * @param bool $defaultLevelUsed
-	 * @return int
+	 * @param ErrorFormatter $formatter
+	 * @return int Error code.
 	 */
 	public function analyse(array $paths, StyleInterface $style, bool $defaultLevelUsed): int
```

Comment:
Maybe the style could carry the `$defaultLevelUsed` information?  It is only relevant to one formatter, but I would now have to pass it everywhere just because it might be required in some.